### PR TITLE
RC vehicle steering indicator & vision

### PR DESCRIPTION
**Pull request by naka123**
#### Summary
SUMMARY: Features "Adds steering indicator to remote controlled vehicle and allows to see through it's camera(s)" 

#### Purpose of change
This change intended to make remote controlled vehicles more useable and controllable

#### Describe the solution
Lightmap processing was changed to use controlled vehicle's cameras and mirrors.
Indication of turning direction was tweaked to account remote-controlled vehicle position.

#### Describe alternatives you've considered
Do not use remote controlled vehicles at all, because in current state its almost impossible to drive it.

#### Additional context
There is video of controlling 1-tile sized remote vehicle inside of research facility building - https://www.reddit.com/r/cataclysmdda/comments/eq8kjz/wip_making_remote_controled_vehicles_usable/

### CODE REVIEW

File: src/cata_tiles.cpp [hunk 1]
```diff
@@ -2572,7 +2572,7 @@ bool cata_tiles::draw_vpart( const tripoint &p, lit_level ll, int &height_3d,
     const bool overridden = override != vpart_override.end();
     // first memorize the actual vpart
     const optional_vpart_position vp = g->m.veh_at( p );
-    if( vp && !invisible[0] ) {
+    if( vp && ( &vp->vehicle() == g->remoteveh_cache || !invisible[0] ) ) {
```
Comment (Qrox) [inline src/cata_tiles.cpp]: Does this mean you can see every part of a remote controlled vehicle even if you cannot normally see them? If that's the case it's not very realistic. People should only be able to see their remotely controlled vehicle if they actually installed a camera system to monitor the parts they want to see.

Comment (naka123) [reply src/cata_tiles.cpp]: As for me, it is same as expirienced driver can "feel" the dimensions of vehicle, even he can't see  some of its parts.

Comment (naka123) [reply src/cata_tiles.cpp]: Also it's a sort of tradeof between "playability" and realism.

Comment (Qrox) [reply src/cata_tiles.cpp]: An experienced drive may feel the dimension of the vehicle, but the current code also lets one know any change that happens to the vehicle telepathically. It's not that realistic if your remotely controlled vehicle bumps into something and you know immediately it's the left headlight that got knocked off.

Comment (kevingranade) [reply src/cata_tiles.cpp]: You can develop the ability to "feel" the dimensions of the vehicle in game, no need to arbitrarally make the vehicle count as transparent.

**Comment by naka123** [comment]
> This needs to be limited to RC vehicles that have a camera and a remote with a screen.
> Ideally it would be directional, but we can look into adding that later.

"remote vehicle controller" recipe contains "large LCD screen", so it looks like all vehicle (not RC car) remote controlles have a screen. as for CBM - Night/Infrared/Scent Vision CBM can "display" video to user without any screens, so Remote Vehicle Controller CBM should alse be able to do so.

**Comment by ymber** [comment]
Fixes #36965. How hard would it be to have the camera automatically follow the vehicle?

**Comment by naka123** [comment]
> Fixes #36965. How hard would it be to have the camera automatically follow the vehicle?

simple "hackish" patch for this - [https://gist.github.com/naka123/00fbfc7be2571c025d08b3b2c27a9c55](url)

**Comment by naka123** [comment]
> Fixes #36965. How hard would it be to have the camera automatically follow the vehicle?

simple "hackish" patch for this - [https://gist.github.com/naka123/00fbfc7be2571c025d08b3b2c27a9c55](url)

**Comment by ZhilkinSerg** [comment]
Feel free to reopen if you want working on it again.

**Pull request closed without merging**